modules/blake3/1.5.4.bcr.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,69 @@
+--- /dev/null
++++ BUILD.bazel
+@@ -0,0 +1,66 @@
++load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
++
++cc_library(
++    name = "blake3",
++    srcs = [
++        "c/blake3.c",
++        "c/blake3_dispatch.c",
++        "c/blake3_portable.c",
++    ] + select({
++        "@bazel_tools//src/conditions:linux_x86_64": [
++            "c/blake3_avx2_x86-64_unix.S",
++            "c/blake3_avx512_x86-64_unix.S",
++            "c/blake3_sse2_x86-64_unix.S",
++            "c/blake3_sse41_x86-64_unix.S",
++        ],
++        "@bazel_tools//src/conditions:linux_aarch64": [
++            "c/blake3_neon.c",
++        ],
++        "@bazel_tools//src/conditions:windows_x64": [
++            "c/blake3_avx2_x86-64_windows_msvc.asm",
++            "c/blake3_avx512_x86-64_windows_msvc.asm",
++            "c/blake3_sse2_x86-64_windows_msvc.asm",
++            "c/blake3_sse41_x86-64_windows_msvc.asm",
++        ],
++        "@bazel_tools//src/conditions:windows_arm64": [
++            "c/blake3_neon.c",
++        ],
++        "@bazel_tools//src/conditions:darwin_arm64": [
++            "c/blake3_neon.c",
++        ],
++        "//conditions:default": [],
++    }),
++    hdrs = [
++        "c/blake3.h",
++        "c/blake3_impl.h",
++    ],
++    copts = select({
++        "@bazel_tools//src/conditions:linux_x86_64": [],
++        "@bazel_tools//src/conditions:linux_aarch64": [
++            "-DBLAKE3_USE_NEON=1",
++        ],
++        "@bazel_tools//src/conditions:windows_x64": [],
++        "@bazel_tools//src/conditions:windows_arm64": [
++            "-DBLAKE3_USE_NEON=1",
++        ],
++        "@bazel_tools//src/conditions:darwin_arm64": [
++            "-DBLAKE3_USE_NEON=1",
++        ],
++        "//conditions:default": [
++            "-DBLAKE3_NO_SSE2",
++            "-DBLAKE3_NO_SSE41",
++            "-DBLAKE3_NO_AVX2",
++            "-DBLAKE3_NO_AVX512",
++        ],
++    }),
++    includes = ["c"],
```

Comment:
Wouldn't this break existing usages of `c/blake3.h` like https://cs.opensource.google/bazel/bazel/+/master:src/main/native/blake3_jni.cc;l=19?q=%22blake3.h%22&ss=bazel

Reply:
It is a breaking change, however it _looks_ like the original library's intention is to allow using <blake3.h> [[1](https://github.com/BLAKE3-team/BLAKE3/blob/df610ddc3b93841ffc59a87e3da659a15910eb46/c/CMakeLists.txt#L292), [2](https://github.com/BLAKE3-team/BLAKE3/blob/df610ddc3b93841ffc59a87e3da659a15910eb46/c/README.md?plain=1#L9)], so stripping the prefix would make it more compatible with non-bazel environments. There are other options including keep as-is, move the BUILD file into the `c` directory, etc. Let me know how you would like to proceed.

Reply:
Can we keep both "." and "c" here? Will there be any conflict?

Reply:
It does seem "c/blake3.h" isn't used widely: https://sourcegraph.com/search?q=context:global+%22%5C%22c/blake3.h%22+file:.*.cc&patternType=keyword&sm=0

But if we can not break them, it's better. I think keep "." in `includes` could probably work.

Reply:
I replaced `include` with `strip_include_prefix`, as the only exported header file is `blake3.h` and there is no need to add root to the include path. In this way, it seems that including both `blake3.h` and `c/blake3.h` works, and the user is expected to eventually remove the prefix. I also moved `blake3_impl.h` from `hdrs` to `srcs` as the file is internal. Please see if this looks good. Thanks!